pkg/hyper/container.go

```diff
@@ -156,15 +156,15 @@ func (h *Runtime) ListContainers(filter *kubeapi.ContainerFilter) ([]*kubeapi.Co
 		}
 
 		if filter != nil {
-			if filter.Id != nil && c.ContainerID != filter.GetId() {
+			if filter.Id != "" && c.ContainerID != filter.Id {
```

Comment:
it's necessary to check `filter.Id != ""` ,because `filter.Id == ""` means match all containerID in list